src/smartcontracts/typesystem/types.ts

```diff
@@ -194,7 +247,7 @@ export abstract class PrimitiveValue extends TypedValue {
 }
 
 export function isTyped(value: any) {
-    return value instanceof TypedValue;
+    return value.belongsToTypesystem;
```

Comment:
```suggestion
    return value.belongsToTypesystem !== undefined;
```
`isTyped` should return `true`/`false` rather than a function or `undefined`.